### PR TITLE
Refactor exporter for anomaly task and fix a bug with exportable code

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary
Fix CVS-144836. Refactor a bit exporter to look similar to other tasks.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/exporter/anomaly.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            ("model_info", "normalization_scale"): self.normalization_scale,
+            ("model_info", "orig_height"): self.orig_height,
+            ("model_info", "orig_width"): self.orig_width,
+            ("model_info", "image_shape"): (self.orig_height, self.orig_width),
+            ("model_info", "labels"): "Normal Anomaly",
+            ("model_info", "model_type"): "AnomalyDetection",
+            ("model_info", "task"): self.task.value,
+        }
+
+    def to_openvino(
```
Comment (sovrasov) [inline src/otx/core/exporter/anomaly.py]: Are .to* methods somehow different from the NativeExporte?

**Comment by harimkang** [comment]
@ashwinvaidya17 Can you take a look at this PR? and I have one question, why didn't that issue show up in the integration test?

File: src/otx/core/model/anomaly.py [hunk 2]
```diff
@@ -402,22 +330,17 @@ def export(
...
-            image_shape=image_shape,
-            image_threshold=self.image_threshold.value.cpu().numpy().tolist(),
-            pixel_threshold=self.pixel_threshold.value.cpu().numpy().tolist(),
-            task=self.task,
-            mean_values=mean_values,
-            scale_values=scale_values,
-            normalization_scale=max_val - min_val,
-        )
-        return exporter.export(
+        if export_format == OTXExportFormatType.OPENVINO:
```
Comment (eunwoosh) [inline src/otx/core/model/anomaly.py:342]: When I tried to make exportable_code, it works on anomaly classification. Are there other reasons not to support it?

File: src/otx/core/model/anomaly.py [hunk 3]
```diff
@@ -384,12 +295,29 @@ def _customize_outputs(
...
+            normalization_scale=max_val - min_val,
+        )
+
     def export(
         self,
         output_dir: Path,
         base_name: str,
         export_format: OTXExportFormatType,
         precision: OTXPrecisionType = OTXPrecisionType.FP32,
+        to_exportable_code: bool = False,
```
Comment (eunwoosh) [inline src/otx/core/model/anomaly.py:328]: Could you add docstring of this argument?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work. I left some comments. Please take a look.

**Comment by harimkang** [comment]
@kprokofi The integration test locally throws 'FAILED' due to that issue, but it seems to be passing in CI. We'll need to do a general check to make sure our export-related integration tests are working locally.

File: src/otx/core/model/anomaly.py (hunk 2 above)
Comment (kprokofi) [reply src/otx/core/model/anomaly.py:342]: The answer is simple. We don't have demo with anomaly. It produces zip archive of course, but we can't run anomaly there

**Review by sovrasov — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Pull request merged**